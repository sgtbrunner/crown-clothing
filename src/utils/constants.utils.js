// INITIAL STATE
export const DEFAULT_INITIAL_INPUT_FIELD_STATE = {
  isValid: false,
  value: '',
  showError: false,
};

// FORM STRINGS
export const DISPLAY_NAME = 'displayName';
export const EMAIL = 'email';
export const PASSWORD = 'password';
export const CONFIRM_PASSWORD = 'confirmPassword';

// ERROR MESSAGES
export const DISPLAY_NAME_ERROR_MESSAGE =
  'Display name must have at least 3 characters';
export const EMAIL_ERROR_MESSAGE = 'Please enter a valid email address';
export const PASSWORD_ERROR_MESSAGE =
  'Your password must be at least 6 characters long';
export const CONFIRM_PASSWORD_ERROR_MESSAGE =
  'Password and confirm password do not match';
export const UNKNOWN_ERROR =
  'An unknown error occurred. Please try again later!';

// OTHER
export const EMPTY_CART_MESSAGE = 'Your cart is empty...';
export const BUTTON = 'button';
