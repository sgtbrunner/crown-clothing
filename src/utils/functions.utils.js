const EMAIL_REGEX_KEY = /^([\w.%+-]+)@([\w-]+\.)+([\w]{2,})$/i;

export const capitalizeFirstLetter = (string) =>
  string[0].toUpperCase() + string.slice(1);

export const isEmailValid = (email) =>
  EMAIL_REGEX_KEY.test(String(email).toLowerCase());
