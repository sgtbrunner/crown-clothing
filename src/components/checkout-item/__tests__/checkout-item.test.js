import React from 'react';
import { screen, render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

import { BUTTON } from '../../../utils/constants.utils';
import CheckoutItem from '../checkout-item.component';

const mockStore = configureMockStore();

const props = {
  cartItem: {
    id: 1,
    price: 10,
    name: 'Item',
    quantity: 2,
    imageUrl: 'https://www.myimage.com/item',
  },
  clearItemFromCart: jest.fn(),
  addItem: jest.fn(),
  removeItem: jest.fn(),
};

const mountComponent = () =>
  render(
    <Provider store={mockStore({})}>
      <CheckoutItem {...props} />
    </Provider>
  );

describe('CheckoutItem component', () => {
  const { name, quantity, price, imageUrl } = props.cartItem;

  it('should match snapshot', () => {
    const { container } = mountComponent();

    expect(container).toMatchSnapshot();
  });

  it('should render checkout item image', () => {
    mountComponent();

    const image = screen.getByAltText('checkout-item');
    expect(image).toBeInTheDocument();
    expect(image).toHaveProperty('src', imageUrl);
  });

  it('should render checkout item text', () => {
    mountComponent();

    const itemText = screen.getByText(name);
    expect(itemText).toBeInTheDocument();
  });

  it('should render decrease item count button', () => {
    mountComponent();

    const decreaseCountButton = screen.getByRole(BUTTON, { name: '❮' });
    expect(decreaseCountButton).toBeInTheDocument();
  });

  it('should render checkout item quantity', () => {
    mountComponent();

    const itemQuantity = screen.getByText(quantity.toString());
    expect(itemQuantity).toBeInTheDocument();
  });

  it('should render increase item count button', () => {
    mountComponent();

    const increaseCountButton = screen.getByRole(BUTTON, { name: '❯' });
    expect(increaseCountButton).toBeInTheDocument();
  });

  it('should render checkout item price', () => {
    mountComponent();

    const itemPrice = screen.getByText(`$${price}`);
    expect(itemPrice).toBeInTheDocument();
  });

  it('should render clear items button', () => {
    mountComponent();

    const clearItemsButton = screen.getByRole(BUTTON, { name: '✖' });
    expect(clearItemsButton).toBeInTheDocument();
  });
});
