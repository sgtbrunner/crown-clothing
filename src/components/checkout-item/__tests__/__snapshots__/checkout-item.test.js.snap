// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutItem component should match snapshot 1`] = `
<div>
  <div
    class="sc-bdnxRM dAKMoO"
  >
    <div
      class="sc-gtsrHT hFUtMm"
    >
      <img
        alt="checkout-item"
        src="https://www.myimage.com/item"
      />
    </div>
    <span
      class="sc-dlnjwi dQblgh"
    >
      Item
    </span>
    <span
      class="sc-dlnjwi sc-hKFxyN dQblgh cywhXn"
    >
      <div
        role="button"
      >
        ❮
      </div>
      <span
        class="sc-dlnjwi dQblgh"
      >
        2
      </span>
      <div
        role="button"
      >
        ❯
      </div>
    </span>
    <span
      class="sc-dlnjwi dQblgh"
    >
      $10
    </span>
    <div
      class="sc-eCApnc btQqcL"
      role="button"
    >
      ✖
    </div>
  </div>
</div>
`;
